File: tests/libjournal/ansi_null.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <j2/journal.h>
// support
#include <cassert>


// type aliases
using ansi_t = pyre::journal::ansi_t;


#include <iostream>
// exercise the trivial device
int main(int argc, char *argv[]) {
    // verify that the null color table support all possible requests
    auto & strange = ansi_t::null["a-very-unlikely-color-name"];
    // and that the color sequences it returns are empty strings
    assert (strange.empty());

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/ansi_null.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


def test():
    """
    Check a few of the canonical X11 color names
    """
    # access the color map
    from j2.ANSI import ANSI

    # ask for a very strange color name
    assert ANSI.x11("a-very-unlikely-color-name") == ""

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
